common/src/stack/command/stack/argument_processors/repo.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+
+from stack.exception import ArgNotFound
+
+class RepoArgProcessor:
+	REQUIRED_REPO_COLUMNS = [
+		'name',
+		'alias',
+		'url',
+	]
+
+	OPTIONAL_REPO_COLUMNS = [
+		'autorefresh',
+		'assumeyes',
+		'type',
+		'is_mirrorlist',
+		'gpgcheck',
+		'gpgkey',
+		'os',
+		'pallet_id',
+	]
+
+	REPO_COLUMNS = REQUIRED_REPO_COLUMNS + OPTIONAL_REPO_COLUMNS
+
+	def insert_repo(self, name, alias, url, **kwargs):
+		'''
+		Insert a repo with optional kwargs into the repos table.
+		This works like add pallet - if you try to add a repo that already exists, silently succeed
+		'''
+		data = {}
+		for key in self.OPTIONAL_REPO_COLUMNS:
+			if key in kwargs:
+				data[key] = kwargs[key]
+
+		where_clause = ' AND '.join(f'{col}=%s' for col in self.REQUIRED_REPO_COLUMNS + list(data))
+		select_stmt = f'''id FROM repos WHERE {where_clause}'''
+
+		if len(self.db.select(select_stmt, (name, alias, url) + tuple(data.values()))) != 0:
+			return
```

Comment:
This looks like if you added an existing repo a second time, but changed a bit of the data, such as the `assumeyes`, you would end up with a duplicate repo (with the one different value) in the DB, because the `select` wouldn't match. Is this the desired behavior?

Reply:
Nope, corrected the behavior and re-wrote.  Add repo $existing_repo is now an error.